src/broadcast/macos/macosmediaplayerservice.mm

```diff
@@ -0,0 +1,128 @@
+    }
+}
+}
+
+void MacOSMediaPlayerService::slotBroadcastCurrentTrack(TrackPointer pTrack) {
+    MPNowPlayingInfoCenter* center = [MPNowPlayingInfoCenter defaultCenter];
+    NSMutableDictionary* nowPlayingInfo = [[NSMutableDictionary alloc] init];
+
+    if (pTrack == nullptr) {
+        [center setPlaybackState:MPNowPlayingPlaybackStatePaused];
+    } else {
+        // Request cover art, possibly loaded asynchronously
+        if (@available(macOS 10.13.2, *)) {
+            CoverArtCache* coverArtCache = CoverArtCache::instance();
+            CoverInfo coverInfo = pTrack->getCoverInfoWithLocation();
+            QPixmap pixmap = coverArtCache->tryLoadCover(
+                this,
+                coverInfo,
+                coverArtSize,
+                CoverArtCache::Loading::Default);
+            if (!pixmap.isNull()) {
+                setCoverArt(nowPlayingInfo, pixmap);
+            }
+        }
+
+        // Retrieve general track metadata
+        NSURL* url = [NSURL fileURLWithPath:pTrack->getLocation().toNSString()];
+        NSString* title = pTrack->getTitle().toNSString();
+        NSString* artist = pTrack->getArtist().toNSString();
+        NSString* album = pTrack->getAlbum().toNSString();
+        NSString* albumArtist = pTrack->getAlbumArtist().toNSString();
+        NSNumber* duration = @(pTrack->getDuration());
+
+        [nowPlayingInfo setObject:@(MPNowPlayingInfoMediaTypeAudio)
+                           forKey:MPNowPlayingInfoPropertyMediaType];
+        [nowPlayingInfo setObject:url forKey:MPNowPlayingInfoPropertyAssetURL];
+        [nowPlayingInfo setObject:@(false) forKey:MPNowPlayingInfoPropertyIsLiveStream];
+        [nowPlayingInfo setObject:title forKey:MPMediaItemPropertyTitle];
+        [nowPlayingInfo setObject:artist forKey:MPMediaItemPropertyArtist];
+        [nowPlayingInfo setObject:album forKey:MPMediaItemPropertyAlbumTitle];
+        [nowPlayingInfo setObject:albumArtist forKey:MPMediaItemPropertyAlbumArtist];
+        [nowPlayingInfo setObject:duration forKey:MPMediaItemPropertyPlaybackDuration];
+
+        // TODO: Use actual play time and (relative) tempo for rate
+        [nowPlayingInfo setObject:@(0.0) forKey:MPNowPlayingInfoPropertyElapsedPlaybackTime];
+        [nowPlayingInfo setObject:@(1.0) forKey:MPNowPlayingInfoPropertyDefaultPlaybackRate];
+        [nowPlayingInfo setObject:@(1.0) forKey:MPNowPlayingInfoPropertyPlaybackRate];
+
+        [center setPlaybackState:MPNowPlayingPlaybackStatePlaying];
+    }
+
+    [center setNowPlayingInfo:nowPlayingInfo];
+}
+
+void MacOSMediaPlayerService::slotCoverFound(const QObject* pRequestor,
+                                             const CoverInfo& coverInfo,
+                                             const QPixmap& pixmap,
+                                             mixxx::cache_key_t requestedImageHash,
+                                             bool coverInfoUpdated) {
+    if (pRequestor != this) {
```

Comment:
Yes, it does happen, I had to add this since the cover art slot would otherwise get flooded with cover art requested by other parts of the program (e.g. the library) and then incorrectly sent to the now playing info.